src/freertos_drivers/net_cc32xx/CC32xxSocket.cxx

```diff
@@ -326,6 +326,7 @@ int CC32xxSocket::accept(int socket, struct sockaddr *address,
     int reserved = reserve_socket();
     if (reserved < 0)
     {
+        sl_Close(result);
```

Comment:
FYI: bugs like this can usually be avoided by using RAII classes like unique_ptr.
Any code where you have the same command repeated more than once before intermediate "return" statements should be a candidate considered for refactoring into an RAII class. With c++11 features like lambda functions this should be really painless.